sensorlib/src/sensor/SemanticLidarSensor.py

```diff
@@ -290,19 +301,22 @@ def compute_instantaneous_actor_id_association(self, hitpoints, scene_objects):
         # Compute nearest neighbor for each hitpoint
         hitpoints_in_map_frame = []
         for hitpoint in hitpoints:
-            new_pos = np.add(hitpoint, self.__sensor.position)
+            sensor_location = self.__sensor.carla_sensor.get_location()
+            new_pos = np.add(hitpoint, np.array([sensor_location.x, sensor_location.y, sensor_location.z]))
             hitpoints_in_map_frame.append(new_pos)
 
         matching_nearest_neighbor_ids = self.compute_closest_object_id_list(hitpoints_in_map_frame, scene_objects,
                                                           self.__simulated_sensor_config["geometry_reassociation"][
                                                               "geometry_association_max_dist_in_meters"])
 
-        association = zip(hitpoints, matching_nearest_neighbor_ids)
+        association = zip(hitpoints_in_map_frame, matching_nearest_neighbor_ids)
```

Comment:
In order to assign hitpoints to closes objects, both need to be in a same frame of reference. hitpoints are reported in sensor's frame. Therefore the library converts the hitpoints into map frame to correctly assign the actor_ids. 
If the use wants sensor_centric hitpoint data, there if a configurable parameter that gives that result, which is by default off right now